### PR TITLE
HelmChartInflationGenerator: add tests for empty documents and splitting

### PR DESCRIPTION
**Pull request by khrisrichardson**
Addresses https://github.com/kubernetes-sigs/kustomize/issues/4914

@natasha41575 I reverted the update to  [HelmChartInflationGenerator.go](https://github.com/kubernetes-sigs/kustomize/compare/master...khrisrichardson:plugin/builtin/helmchartinflationgenerator/split?expand=1#diff-57d88298737665fc6e6ab73c724dc75158356f59786b9ce2335c150945b38f2fR261), since after testing it became evident that only the `render-helm-chart ` function in the `kpt-functions-catalog` was susceptible (i.e - it should be refactored to sync with the updates to the kustomize plugin).

Nevertheless, I included a local chart to simulate both the [splitting](https://github.com/kubernetes-sigs/kustomize/compare/master...khrisrichardson:plugin/builtin/helmchartinflationgenerator/split?expand=1#diff-8469aa0290dbd953d8dbe752c31312243cfb6bc00049951139680c86e9577ca4R9-R10) and [empty](https://github.com/kubernetes-sigs/kustomize/compare/master...khrisrichardson:plugin/builtin/helmchartinflationgenerator/split?expand=1#diff-fc367c47601d54f23b6ecf2f51ec43467deec08a0d8834710175460de04a65b6R20-R27) document issues that were encountered with `render-helm-chart`.

I also included tests for some remote helm charts that exhibited either problem: [bitnami/redis-cluster](https://github.com/kubernetes-sigs/kustomize/compare/master...khrisrichardson:plugin/builtin/helmchartinflationgenerator/split?expand=1#diff-a5f02292d09cefaf3346f1bed0464c9a0aac016e1c00cdfe4e3e95a83f104d79R722-R749), [rook/rook-ceph](https://github.com/kubernetes-sigs/kustomize/compare/master...khrisrichardson:plugin/builtin/helmchartinflationgenerator/split?expand=1#diff-a5f02292d09cefaf3346f1bed0464c9a0aac016e1c00cdfe4e3e95a83f104d79R751-R775), and [smallstep/step-certificates](https://github.com/kubernetes-sigs/kustomize/compare/master...khrisrichardson:plugin/builtin/helmchartinflationgenerator/split?expand=1#diff-a5f02292d09cefaf3346f1bed0464c9a0aac016e1c00cdfe4e3e95a83f104d79R777-R801)

### CODE REVIEW

**Review by natasha41575 — COMMENTED** [review]
Thanks so much for this PR!! Really appreciate both the investigation and tests; more tests are always welcome In the interest of not rushing you for time, I'm going to include the reverted change as part of https://github.com/kubernetes-sigs/kustomize/pull/4926 - which means you don't need to worry about getting this merged before we release.

I think the tests look good, but I do have a small question below.

File: plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator_test.go [hunk 1]
```diff
@@ -579,3 +580,222 @@ valuesInline:
...
+apiVersion: builtin
+kind: HelmChartInflationGenerator
+metadata:
+  name: step-certificates
+name: step-certificates
+repo: https://smallstep.github.io/helm-charts
+version: 1.23.0
+releaseName: certificates
+`)
+	th.AssertActualEqualsExpected(rm, string(testData))
```
[the remark this answers is not in the document]
Comment (khrisrichardson) [reply plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator_test.go]: @natasha41575 I made the changes you suggested
